src/server/game/Entities/Player/Player.cpp

```diff
@@ -20782,7 +20782,10 @@ void Player::Say(std::string_view text, Language language, WorldObject const* /*
 
     WorldPacket data;
     ChatHandler::BuildChatPacket(data, CHAT_MSG_SAY, language, this, this, _text);
-    SendMessageToSetInRange(&data, sWorld->getFloatConfig(CONFIG_LISTEN_RANGE_SAY), true);
+    
```

Comment:
In Player.cpp you have three identical fragments:

1. call SendDirectMessage
2. creating instance of ChatMessageDistDeliverer
3. call Cell::VisitWorldObjects

According to DRY prinicple (do not repeat yourself) - you should create new member-function with meaningful name